src/syscall/syscall_wasi.go

```diff
@@ -0,0 +1,123 @@
+// +build wasi
```

Comment:
This file is identical to syscall_baremetal.go (except for this build tag), so it should be possible to merge the two.